src/components/ViewArticles.jsx

```diff
@@ -3,33 +3,57 @@ import React, { useState, useEffect } from 'react';
 import { getAllArticles } from './utils/api';
 function ViewArticles() {
   const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
   
   useEffect(() => {
     getAllArticles()
       .then((articlesFromApi) => {
         console.log(articlesFromApi);
         setArticles(articlesFromApi);
+        setIsLoading(false);
       })
       .catch((err) => console.error(err));
   }, []);
 
+  if (isLoading)
   return (
-    <div>
-      <h1>View Articles</h1>
-      {articles.map(article => (
-        
-        <div key={article.article_id}>
-          <img src={article.article_img_url} alt={`A picture of ${article.title}`} />
-          <h2>{article.title}</h2>
-       
-          <p>Written by {article.author}</p>
-
-          <p>Posted on {new Date(article.created_at).toLocaleDateString('en-GB', { day: 'numeric', month: 'long', year: 'numeric' })}</p>          
-          <p>{article.votes} votes</p>
-          <p>{article.comment_count} comments</p>
-          Category: <a href="">coding</a>
+    <div className="loading-spinner">
```

Comment:
try making this a resuable component - easier than having to copy and paste this every time you want loading functionality